### PR TITLE
Improved error handling for reference methods

### PR DESCRIPTION
**Pull request by Erfanm83**
**Summary:**
Hi @dnnrly,

Following up on our previous discussions and improvements (#49), I have taken another look at the codebase and identified areas where the use of `panic()` might lead to unexpected crashes. To improve the robustness of the application, I have replaced the `panic()` calls with proper error handling and logging, ensuring that edge cases are handled gracefully. These changes should make the application more resilient and enhance user experience.

**Changes made:**
- Replaced instances of `panic()` with error logging and proper fallback mechanisms in `httpref.go`.
- Added checks to ensure values are valid before accessing them to avoid runtime errors.
- Enhanced error messages to provide more clarity during failures.

Please let me know if there are any additional amendments you would like or if there are other areas of the code that could benefit from similar treatment.

Thank you again for reviewing this and providing feedback, as always!

Best,  
@Erfanm83

### CODE REVIEW

File: httpref.go [hunk 1]
```diff
@@ -40,7 +41,13 @@ func (r References) InRange(code string) References {
 	for _, v := range r {
 		if strings.Contains(v.Name, "-") {
 			parts := strings.Split(v.Name, "-")
-			if code >= parts[0] && code <= parts[len(parts)-1] {
+			if len(parts) != 2 {
+				fmt.Printf("Invalid range format for reference name: %s\n", v.Name)
+				continue
+			}
```
Comment (dnnrly) [inline httpref.go:47]: We can probably skip this as the tests will catch this problem - the command will crash in the acceptance tests.

File: httpref.go [hunk 2]
```diff
@@ -40,7 +41,13 @@ func (r References) InRange(code string) References {
...
+
+			start, end := parts[0], parts[1]
+			if code >= start && code <= end {
```
Comment (dnnrly) [inline httpref.go:50]: There should be an additional test for this behaviour if something is changing.

File: httpref.go [hunk 3]
```diff
@@ -62,6 +69,7 @@ func (r References) Titles() References {
 	return results
 }
 
+// Search looks for references that contain the search term in their fields
```
Comment (dnnrly) [inline httpref.go:72]: :+1:

File: view.go [hunk 4]
```diff
@@ -26,15 +26,22 @@ func (r Reference) Summarize(style lipgloss.Style) string {
...
-		// hoping this doesn't happen as most commands here suceed without issue
-		panic(err)
+		fmt.Fprintf(os.Stderr, "Failed to update term renderer: %v\n", err)
+		return "Error rendering description"
 	}
 	description, err := descriptionStyle.Render(r.Description)
 	if err != nil {
-		panic(err)
+		fmt.Fprintf(os.Stderr, "Failed to render description: %v\n", err)
+		return "Error rendering description"
```
Comment (dnnrly) [inline view.go:44]: I think I would prefer returning `(string, error)` here. Then we can handle the failure by exiting with a non-zero code properly.

File: view.go [hunk 5]
```diff
@@ -46,6 +53,12 @@ func init() {
 }
 
 func renderStyles() {
+	defer func() {
+		if rec := recover(); rec != nil {
+			fmt.Fprintf(os.Stderr, "An error occurred during renderStyles: %v\n", rec)
+		}
+	}()
```
Comment (dnnrly) [inline view.go:60]: I like this catch-all. Would it possible to add some text that directs the user to raise an issue in Github? Maybe a pre-populated link?

File: view.go [hunk 6]
```diff
@@ -63,8 +76,8 @@ func renderStyles() {
 
 	r, err := updateTermRendered(resultStyle)
 	if err != nil {
-		// hoping this doesn't happen as most commands here suceed without issue
-		panic(err)
+		fmt.Fprintf(os.Stderr, "Failed to update term renderer in renderStyles: %v\n", err)
+		return
```
Comment (dnnrly) [inline view.go:80]: Maybe return an error here too.

File: view.go [hunk 7]
```diff
@@ -84,6 +97,13 @@ func updateTermRendered(style lipgloss.Style) (*glamour.TermRenderer, error) {
 }
 
 func PrintResultsWithStyle(results References, rootStyle lipgloss.Style) {
+	defer func() {
+		if rec := recover(); rec != nil {
+			fmt.Fprintf(os.Stderr, "An error occurred during PrintResultsWithStyle: %v\n", rec)
+			os.Exit(1)
+		}
+	}()
```
Comment (dnnrly) [inline view.go:105]: Can the different recovery blocks be combined? I'm not sure we should be using this for regular error handling.

**Comment by dnnrly** [comment]
Once again, another strong contribution!

File: httpref.go (hunk 1 above)
Comment (Erfanm83) [reply httpref.go:47]: > We can probably skip this as the tests will catch this problem - the command will crash in the acceptance tests.

well I will remove this from httpref.go

File: httpref.go (hunk 2 above)
Comment (Erfanm83) [reply httpref.go:50]: > There should be an additional test for this behaviour if something is changing.

well I will add some sort of tests for an edge case at httpref_test.go file

**Review by Erfanm83 — COMMENTED** [review]
### Pull Request Title: Improved Error Handling for Reference Methods

**Summary:**
Hi @dnnrly,

Thank you for the detailed review. I've made the following changes based on your feedback:

**Changes made:**
1. Removed the `len(parts) != 2` check in `httpref.go`, as suggested, to allow tests to handle any malformed input scenarios.
2. Added additional tests to ensure the behavior of range matching is fully covered and verified.
3. Modified the `view.go` function to return `(string, error)` instead of using `panic()`, allowing the errors to be properly handled with a non-zero exit code.
4. Updated recovery messages in `view.go` to direct users to raise an issue on GitHub, with a link provided.
5. Changed the `renderStyles()` function to return an error instead of just logging to `os.Stderr` to better align with the overall error-handling strategy.
6. Refactored recovery blocks in `view.go` to reduce redundancy and centralize the recovery mechanism.

Please let me know if there are any additional changes you would like to see.

Thank you for guiding me through this process!

If possible, could you add the `hacktoberfest-accepted` label to this PR? I'd be really grateful as I'm aiming for my Hacktoberfest badge. Thanks again for your help and feedback!

Best,  
@Erfanm83

**Comment by Erfanm83** [comment]
hey @dnnrly Can you please checkout my changes and tell me where should I Change ?
It has been a long time and I got mo reveiw !
Best,
Erfan

**Comment by dnnrly** [comment]
Apologies, I've been away recently so haven't been able to look this over until now.
